packages/theme-check-common/src/test/MockFileSystem.ts

```diff
@@ -46,32 +48,37 @@ export class MockFileSystem implements AbstractFileSystem {
   async stat(uri: string): Promise<FileStat> {
     const relativePath = this.rootRelative(uri);
     const source = this.mockTheme[relativePath];
-    if (source === undefined) {
-      throw new Error('File not found');
+    if (source) {
+      return {
+        type: FileType.File,
+        size: source.length ?? 0,
+      };
     }
-    return {
-      type: FileType.File,
-      size: source.length ?? 0,
-    };
-  }
 
-  private _fileTree: FileTree | null = null;
+    const readdirResult = await this.readDirectory(uri);
+    if (readdirResult) {
+      return {
+        type: FileType.Directory,
+        size: 0, // Size is not applicable for directories
+      };
+    }
+
+    throw new Error(`File not found: ${uri} for ${this.rootUri}`);
+  }
 
   private get fileTree(): FileTree {
-    if (!this._fileTree) {
-      this._fileTree = {};
-      for (const [relativePath, source] of Object.entries(this.mockTheme)) {
-        const segments = relativePath.split('/');
-        let current = this._fileTree;
-        for (let i = 0; i < segments.length - 1; i++) {
-          const segment = segments[i];
-          current[segment] ??= {};
-          current = current[segment] as FileTree;
-        }
-        current[segments[segments.length - 1]] = source;
+    const result: FileTree = {};
+    for (const [relativePath, source] of Object.entries(this.mockTheme)) {
+      const segments = relativePath.split('/');
+      let current = result;
+      for (let i = 0; i < segments.length - 1; i++) {
+        const segment = segments[i];
+        current[segment] ??= {};
+        current = current[segment] as FileTree;
       }
+      current[segments[segments.length - 1]] = source;
     }
-    return this._fileTree;
+    return result;
```

Comment:
Had to remove memoization because of a test that mutated this.mockTheme. It's only for tests, perf doesn't matter :)